src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -977,7 +977,9 @@ public Response publishDataset(@PathParam("id") String id, @QueryParam("type") S
                 logger.fine("MT: " + ds.getModificationTime());
                 logger.fine("PIT: " + ds.getPermissionIndexTime());
                 logger.fine("PMT: " + ds.getPermissionModificationTime());
-                logger.fine("ITMT: " + (ds.getIndexTime().compareTo(ds.getModificationTime()) <= 0));
+                if (ds.getIndexTime() != null && ds.getModificationTime() != null) {
```

Comment:
If either of these is `null` a NullPointerException will be thrown.

Reply:
Yes - non-null checks are needed here